autotest.pm

```diff
@@ -463,7 +463,7 @@ sub loadtestdir ($dir) {
     $dir =~ s/^\Q$bmwqemu::vars{CASEDIR}\E\/?//;    # legacy where absolute path is specified
     $dir = join('/', $bmwqemu::vars{CASEDIR}, $dir);    # always load from casedir
     die "'$dir' does not exist!\n" unless -d $dir;
-    loadtest($_) for (glob "$dir/*.pm");
+    loadtest($_) for map { s{^\Q$bmwqemu::vars{CASEDIR}/}{}r } (glob "$dir/*.pm");
```

Comment:
So two lines above dir is added and here you remove it again?

Reply:
Your new change cannot work. @okurz was not right.
`glob` needs to get the full path since we cannot rely that the current working directory is that directory.

Tests are failing, and I wonder if you thought about updating the tests according to the acceptance criteria?
The test is still doing a `use lib`, which it shouldn't have to do, and the test checks that a warning is created, while the fix should ensure that there is no warning anymore.

Reply:
Maybe if you create the relative path after "globbing" ? Something like
```
$dir = join('/', $bmwqemu::vars{CASEDIR}, $dir);    # always load from casedir
die "'$dir' does not exist!\n" unless -d $dir;
for my $testfile (glob "$dir/*.pm") {
    (my $relfile = $testfile) =~ s{^\Q$casedir/}{};
    loadtest($relfile);
}
```

Reply:
That for loop is basically the same as `loadtest($_) for map { s{^\Q$bmwqemu::vars{CASEDIR}/}{}r } (glob "$dir/*.pm");`

Reply:
ah i guess you're right, sorry